common/ManagementTestShared/MgmtRecordTestBase.cs

```diff
@@ -0,0 +1,226 @@
+        {
+            return Task.FromResult<object>(null);
+        }
+        protected virtual Task OnOneTimeSetupScenarioAsync()
+        {
+            return Task.FromResult<object>(null);
+        }
+        protected virtual Task OnSetupBaseAsync()
+        {
+            return Task.FromResult<object>(null);
+        }
+        protected virtual Task OnSetupScenarioAsync()
+        {
+            return Task.FromResult<object>(null);
+        }
+        protected virtual Task OnOneTimeTearDownBaseAsync()
+        {
+            return Task.FromResult<object>(null);
+        }
+        protected virtual Task OnOneTimeTearDownScenarioAsync()
+        {
+            return Task.FromResult<object>(null);
+        }
+
+        protected virtual Task OnTearDownBaseAsync()
+        {
+            return Task.FromResult<object>(null);
+        }
+
+        protected virtual Task OnTearDownScenarioAsync()
+        {
+            return Task.FromResult<object>(null);
+        }
+
+        [OneTimeSetUp]
+        public void RunOneTimeSetup()
+        {
+            if (Mode == RecordedTestMode.Live || Debugger.IsAttached)
+            {
+                Logger = new TestLogger();
+            }
+            Recording = new TestRecording(Mode, GetSessionFilePath(), Sanitizer, Matcher);
+            TestEnvironment.SetRecording(Recording);
+            InitializeClients();
+            OnOneTimeSetupBaseAsync();
+            OnOneTimeSetupScenarioAsync();
+        }
+
+        [OneTimeTearDown]
+        public async Task RunOneTimeTearDown()
+        {
+            StopTestRecording();
+            await CleanupResourceGroupsAsync();
+            Logger?.Dispose();
+            Logger = null;
+            await OnOneTimeTearDownScenarioAsync();
+            await OnOneTimeTearDownBaseAsync();
+        }
+
+        [SetUp]
+        public void StartTestRecording()
+        {
+            StopTestRecording();
+            TestContext.TestAdapter test = TestContext.CurrentContext.Test;
+            if (Mode != RecordedTestMode.Live &&
+                test.Properties.ContainsKey("SkipRecordings"))
+            {
+                throw new IgnoreException((string)test.Properties.Get("SkipRecordings"));
+            }
+            Recording = new TestRecording(Mode, GetSessionFilePath(), Sanitizer, Matcher);
+            TestEnvironment.Mode = Mode;
+            TestEnvironment.SetRecording(Recording);
+            InitializeClients();
+            OnSetupBaseAsync();
+            OnOneTimeTearDownScenarioAsync();
+        }
+
+        [TearDown]
+        public void StopTestRecording()
+        {
+            bool save = TestContext.CurrentContext.Result.FailCount == 0;
+#if DEBUG
+            save |= SaveDebugRecordingsOnFailure;
+#endif
+            Recording?.Dispose(save);
+            OnTearDownScenarioAsync();
+            OnTearDownScenarioAsync();
+        }
+
+        protected ResourcesManagementClient GetResourceManagementClient()
```

Comment:
Would like to expose these as singleton properties. So maybe lazy initialize in ctor.